uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveRequestTest.php

```diff
@@ -319,4 +325,359 @@ public function testGetBalanceChangeByAbsenceTypeCanReturnTheBalanceForOnlyPubli
+    $publicHoliday = new PublicHoliday();
+    $publicHoliday->date = date('2016-11-14');
+
+    $this->assertNull(LeaveRequest::findPublicHolidayLeaveRequest($contact['id'], $publicHoliday));
+    PublicHolidayLeaveRequestFabricator::fabricate($contact['id'], $publicHoliday);
+
+    $fromDate = date("2016-11-14");
+    $toDate = date("2016-11-15");
+    $fromType = $this->leaveRequestDayTypes['All Day']['name'];
+    $toType = $this->leaveRequestDayTypes['All Day']['name'];
+
+    $expectedResultsBreakdown = [
+      'amount' => 0,
+      'breakdown' => []
+    ];
+
+    // Starting date is a monday, but a public holiday
+    $expectedResultsBreakdown['amount'] += 0;
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-11-14',
+      'amount' => 0,
+      'type' => [
+        'id' => $this->leaveRequestDayTypes['Public Holiday']['id'],
+        'value' => $this->leaveRequestDayTypes['Public Holiday']['value'],
+        'label' => 'Public Holiday'
+      ]
+    ];
+
+    // last day is a tuesday, which is a working day
+    $expectedResultsBreakdown['amount'] += 1.0;
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-11-15',
+      'amount' => 1.0,
+      'type' => [
+        'id' => $this->leaveRequestDayTypes['All Day']['id'],
+        'value' => $this->leaveRequestDayTypes['All Day']['value'],
+        'label' => 'All Day'
+      ]
+    ];
+
+    $expectedResultsBreakdown['amount'] *= -1;
+
+    $result = LeaveRequest::calculateBalanceChange($contact['id'], $fromDate, $fromType, $toDate, $toType);
+    $this->assertEquals($expectedResultsBreakdown, $result);
+  }
+
+  public function testCalculateBalanceChangeForALeaveRequestForAContactWithMultipleWeeks() {
+    $contact = ContactFabricator::fabricate();
+    $periodStartDate = date('Y-01-01');
+    $title = 'Job Title';
+
+    HRJobContractFabricator::fabricate([
+      'contact_id' => $contact['id']
+    ],
+    [
+      'period_start_date' => $periodStartDate,
+      'title' => $title
+    ]);
+
+    // Week 1 weekdays: monday, wednesday and friday
+    // Week 2 weekdays: tuesday and thursday
+    $pattern = WorkPatternFabricator::fabricateWithTwoWeeksAnd31AndHalfHours();
+    ContactWorkPatternFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'pattern_id' => $pattern->id
+    ]);
+
+    $fromDate = '2016-07-31';
+    $toDate = '2016-08-15';
+    $fromType = $this->leaveRequestDayTypes['All Day']['name'];
+    $toType = $this->leaveRequestDayTypes['1/2 AM']['name'];
+
+    $expectedResultsBreakdown = [
+      'amount' => 0,
+      'breakdown' => []
+    ];
+
+    // Start day (2016-07-31), a sunday
+    $expectedResultsBreakdown['breakdown'][] = [
+      'date' => '2016-07-31',
+      'amount' => 0,
+      'type' => [
+        'id' => $this->leaveRequestDayTypes['Weekend']['id'],
+        'value' => $this->leaveRequestDayTypes['Weekend']['value'],
+        'label' => 'Weekend'
```

Comment:
can't this be used here (and on the other arrays with the expected values)? `$this->leaveRequestDayTypes['Weekend']['label']`